x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go

```diff
@@ -180,10 +180,22 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) error {
 		if err != nil {
 			m.Logger().Warn("skipping metrics list from region '%s'", regionName)
 		}
+
+		// retrieve all the details for all the metrics available in the current region
+		regionListMetricsOutput, err := aws.GetListMetricsOutput("*", regionName, m.Period, svcCloudwatch)
+
+		// filter out metrics details for namespaces that are not included in the configuration
+		listMetricsOutput := make([]types.Metric, 0)
```

Comment:
it seems we should add an if in `filterListMetricsOutput` in order to be sure that the metrics belong the current namespace we are iterating

Reply:
You are right. I made this change in https://github.com/elastic/beats/pull/34055/commits/c58d9d267167a7f8758ac03a63e8b3b6dcb7e8e7 . We could technically remove this part now as the if added in this new commit would filter out also the namespaces we don't have in the conf, but I think it's good to explicitly do it here. What do you think? In terms of performance, it does not change much.